### PR TITLE
Replace `time.Now()` with `clock.Clock.Now()` in pkg/workload/workload.go

### PR DESCRIPTION
**Pull request by TusharMohapatra07**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Replaces `time.Now()` to use the Now() function present inside `clock` package.

#### Which issue(s) this PR fixes:
Part of #3380

#### Special notes for your reviewer:
This PR is work-in-progress. 

```release-note
NONE
```

### CODE REVIEW

**Comment by TusharMohapatra07** [comment]
@mbobrovskyi Please review ! I want to know if my approach is right. Post, approval, i'm going to proceed with other refactors

File: test/integration/controller/jobs/job/job_controller_test.go [hunk 1]
```diff
@@ -622,6 +627,7 @@ var _ = ginkgo.Describe("Job controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 					kueue.WorkloadEvicted,
 					metav1.ConditionTrue,
 					kueue.WorkloadEvictedByPreemption, "By test", "evict",
+					fakeClock,
```
Comment (mbobrovskyi) [inline test/integration/controller/jobs/job/job_controller_test.go:630]: I think we can use realClock on integration tests due to we are using Eventually.

File: pkg/workload/workload.go [hunk 2]
```diff
@@ -585,14 +586,14 @@ func SetQuotaReservation(w *kueue.Workload, admission *kueue.Admission) {
 		evictedCond.Status = metav1.ConditionFalse
 		evictedCond.Reason = "QuotaReserved"
 		evictedCond.Message = api.TruncateConditionMessage("Previously: " + evictedCond.Message)
-		evictedCond.LastTransitionTime = metav1.Now()
+		evictedCond.LastTransitionTime = metav1.Now() //here
```
Comment (mbobrovskyi) [inline pkg/workload/workload.go:589]: What is it?

Comment (TusharMohapatra07) [reply pkg/workload/workload.go:589]: That's for my ref, i'll remove them once i work on those

File: test/integration/controller/jobs/job/job_controller_test.go (hunk 1 above)
Comment (TusharMohapatra07) [reply test/integration/controller/jobs/job/job_controller_test.go:630]: Sure, i'll make the necessary changes

**Review by mbobrovskyi — COMMENTED** [review]
Overall LGTM. Please fix SetQuotaReservation as well.

**Comment by TusharMohapatra07** [comment]
@mbobrovskyi Please review

File: test/util/util.go [hunk 3]
```diff
@@ -590,11 +591,13 @@ func ExpectCQResourceReservations(cq *kueue.ClusterQueue, flavor, resource strin
...
 
 func SetQuotaReservation(ctx context.Context, k8sClient client.Client, wl *kueue.Workload, admission *kueue.Admission) error {
+	now := time.Now()
+	fakeClock := testingclock.NewFakeClock(now)
 	wl = wl.DeepCopy()
 	if admission == nil {
 		workload.UnsetQuotaReservationWithCondition(wl, "EvictedByTest", "Evicted By Test", time.Now())
 	} else {
-		workload.SetQuotaReservation(wl, admission)
+		workload.SetQuotaReservation(wl, admission, fakeClock)
```
Comment (mbobrovskyi) [inline test/util/util.go:600]: Also here use realClock.

Comment (TusharMohapatra07) [reply test/util/util.go]: @mbobrovskyi Done!

**Review by mbobrovskyi — APPROVED** [review]
/lgtm
Thank you so much!

**Comment by TusharMohapatra07** [comment]
@mbobrovskyi hey, one test is failing (integration test). What could be done ? is it because realClock ?

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/approve

**Pull request merged**